ALUM/ALUM/Services/ServiceHelper.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  RequestGenerator.swift
+//  ALUM
+//
+//  Created by Yash Ravipati on 4/24/23.
+//
+
+import Foundation
+
+class ServiceHelper {
+    static let shared = ServiceHelper()
+
+    func attachAuthTokenToRequest(request: inout URLRequest) async throws {
+        guard let authToken = try await FirebaseAuthenticationService.shared.getCurrentAuth() else {
+            throw AppError.actionable(.authenticationError, message: "Error getting auth token")
+        }
+        request.setValue("Bearer \(authToken)", forHTTPHeaderField: "Authorization")
+    }
+
+    func createRequest(urlString: String, method: String, requireAuth: Bool) async throws -> URLRequest {
```

Comment:
While making a new service you just have to use `createRequest` with `requireAuth set as boolean